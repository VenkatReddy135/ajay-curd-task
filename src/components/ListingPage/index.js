import React from "react";

import CustomList from "../../shared_components/CustomList";

const EmployeesList = () => {
  const resource = "https://dummy.restapiexample.com/api/v1";

  const modalData = "Are you sure you want to delete?";

  return (
    <CustomList
      listTitle="Employee List"
      textOnNoData="No Employees Found"
      getAllResource={`${resource}/employees`}
      deleteResource={`${resource}/delete`}
      modalData={modalData}
    />
  );
};

export default React.memo(EmployeesList);
