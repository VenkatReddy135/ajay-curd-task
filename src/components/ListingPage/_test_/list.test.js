import React from "react";
import { cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ListingPage from "../index";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  useNavigate: jest.fn(),
}));

describe("Employee List component page", () => {
  const renderedData = renderer.create(<ListingPage />).toJSON();

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ListingPage />, div);
  });

  it("renders a snapshot", () => {
    expect(renderedData).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
