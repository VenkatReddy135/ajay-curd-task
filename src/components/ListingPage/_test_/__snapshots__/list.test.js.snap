// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee List component page renders a snapshot 1`] = `
<div
  className="container border border-info mt-4 mb-4"
>
  <h2>
    Employee List
  </h2>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <input
        className="form-control searchField marginClass"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
    </div>
  </div>
  <table
    className="table marginClass"
  >
    <thead>
      <tr>
        <th>
          Employee Name
        </th>
        <th>
          Employee Age
        </th>
        <th>
          Employee Salary
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="border-0"
          colSpan={3}
        >
          No Employees Found
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="toast-container position-absolute top-0 end-0 p-3"
  />
</div>
`;
